RadarSDK/RadarLocationManager.m

```diff
@@ -165,17 +166,18 @@ - (void)startTrackingWithOptions:(RadarTrackingOptions *)trackingOptions {
     CLAuthorizationStatus authorizationStatus = [self.permissionsHelper locationAuthorizationStatus];
     if (!(authorizationStatus == kCLAuthorizationStatusAuthorizedWhenInUse || authorizationStatus == kCLAuthorizationStatusAuthorizedAlways)) {
         [[RadarDelegateHolder sharedInstance] didFailWithStatus:RadarStatusErrorPermissions];
-
         return;
     }
 
     [RadarSettings setTracking:YES];
     [RadarSettings setTrackingOptions:trackingOptions];
+    [RadarSettings setFallbackTrackingOptions:trackingOptions];
```

Comment:
So I guess the happy path for a first run would be:
```
1. Radar.initialize()
2. GET /config -> if (res.meta.trackingOptions) { // set tracking options }
3. Radar.startTrackingWithOptions(options)
4. Overwrite the remote tracking options with the local tracking options
5. begin tracking based on local tracking options
6. Make a /track request, update tracking options based on res.meta.trackingOptions
```

@jkao @jsani-radar @jrtibbetts Am I getting that right?

Reply:
I think 3/4 are worth clarifying - should 3 be "overwrite local with remote" as the remote options should take precedence, correct? so 5 should be "begin tracking with remote options"?

Reply:
```
1. Radar.initialize() 
2. GET /config -> if (res.meta.trackingOptions) { // set remote tracking options } 
3. Radar.startTracking(clientSideTrackingOptions) -> RadarLocationManager.updateTracking // updates to remote tracking options from step 2
4. Begin tracking based on local tracking options from server
5. Make a /track request, update tracking options based on res.meta.trackingOptions if present + enabled from the server, or fallback to clientSideTrackingOptions in step 3, otherwise
```